src/main/kotlin/com/openlattice/mechanic/upgrades/PostPermissionMigrationUpgrade.kt

```diff
@@ -0,0 +1,160 @@
+package com.openlattice.mechanic.upgrades
+
+import com.hazelcast.query.Predicates
+import com.openlattice.ApiHelpers
+import com.openlattice.authorization.AccessTarget
+import com.openlattice.authorization.Acl
+import com.openlattice.authorization.AclKey
+import com.openlattice.authorization.DbCredentialService
+import com.openlattice.authorization.Permission
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organization.ExternalColumn
+import com.openlattice.organization.ExternalTable
+import com.openlattice.organizations.mapstores.ORGANIZATION_ID_INDEX
+import com.openlattice.postgres.PostgresPrivileges
+import com.openlattice.postgres.external.ExternalDatabaseConnectionManager
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.sql.SQLException
+import java.util.EnumSet
+import java.util.UUID
+import java.util.regex.Pattern
+
+class PostPermissionMigrationUpgrade(
+        toolbox: Toolbox,
+        private val exConnMan: ExternalDatabaseConnectionManager,
+        private val dbCreds: DbCredentialService
+): Upgrade {
+
+    val logger: Logger = LoggerFactory.getLogger(PostPermissionMigrationUpgrade::class.java)
+
+    private val externalColumns = HazelcastMap.EXTERNAL_COLUMNS.getMap(toolbox.hazelcast)
+    private val externalTables = HazelcastMap.EXTERNAL_TABLES.getMap(toolbox.hazelcast)
+    private val externalRoleNames = HazelcastMap.EXTERNAL_PERMISSION_ROLES.getMap(toolbox.hazelcast)
+    private val organizations = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+    private val olToPostgres = mapOf<Permission, Set<PostgresPrivileges>>(
+        Permission.READ to EnumSet.of(PostgresPrivileges.SELECT),
+        Permission.WRITE to EnumSet.of(PostgresPrivileges.INSERT, PostgresPrivileges.UPDATE),
+        Permission.OWNER to EnumSet.of(PostgresPrivileges.ALL)
+    )
+    private val allTablePermissions = setOf(Permission.READ, Permission.WRITE, Permission.OWNER)
+
+    override fun upgrade(): Boolean {
+        val uuidRegex= Pattern.compile(
+            "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+            Pattern.CASE_INSENSITIVE
+        )
+
+        // filtering for a specific org
+        println("Organization to filter: ")
+        val input = readLine()
+        val filterFlag = uuidRegex.matcher(input).matches()
```

Comment:
Same comments about the regex and `readLine()`.

Reply:
Noted, this file will be fixed when I finished cleaning up the rest of this file (there's some minor issue with how to process property types/entity sets that I am almost done fixing).